### PR TITLE
SW-3718 add search input to observations page

### PR DESCRIPTION
**Pull request by karthikbtf**
- search only searches zone names (as per design annotation)
- added a search selector
- search filters will be added later on

Terraware App 2023-06-06 14-21-24

### CODE REVIEW

File: src/redux/features/observations/observationsSelectors.ts [hunk 1]
```diff
@@ -55,7 +55,22 @@ export const selectMergedPlantingSiteObservations = createCachedSelector(
...
 
-// add more selectors for drill down views, as needed
+// search observations (search planting zone name only)
+export const searchObservations = createCachedSelector(
+  (state: RootState, plantingSiteId: number, defaultTimeZone: string, search: string) =>
+    selectMergedPlantingSiteObservations(state, plantingSiteId, defaultTimeZone),
+  (state: RootState, plantingSiteId: number, defaultTimeZone: string, search: string) => search,
+  (observations, search) =>
+    observations?.filter((observation: ObservationResults) => {
+      const regexp = new RegExp(search, 'i');
```
Comment (charlie-tf) [inline src/redux/features/observations/observationsSelectors.ts:65]: This should allow regular expressions to be input into the search input and ran unescaped against the zone names. Other searches use the search API, making this behavior different. Perhaps not worth worrying about, but just noting.

Comment (charlie-tf) [reply src/redux/features/observations/observationsSelectors.ts:65]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions#escaping has a function that could be used if escaping is desired here.

**Review by charlie-tf — APPROVED** [review]
Looks good. A note and suggestion, if wanted, but not necessary to merge.

File: src/redux/features/observations/observationsSelectors.ts (hunk 1 above)
Comment (karthikbtf) [reply src/redux/features/observations/observationsSelectors.ts:65]: I can certainly add the escaping 👍 , fwiw i didn't know regex search was a requirement in our product. you never know, we could have a tech savvy customer..

Comment (karthikbtf) [reply src/redux/features/observations/observationsSelectors.ts:65]: btw, I was going to use "string.toLowerCase().includes(pattern.toLowerCase())" but went with regexp purely for the case-insensitive option. I think I understand the reason for the suggestion now.
If it's less confusing, I can switch back to "string.includes" approach.

Comment (charlie-tf) [reply src/redux/features/observations/observationsSelectors.ts:65]: Oh, the regexp approach isn't confusing. I just wondered whether allowing full regexp searches was what we wanted here. I'm okay with either approach. Regexp with escaping vs. string.includes should give the same result (I _think_), and either one is clear enough

**Review by charlie-tf — APPROVED** [review]
👍

**Pull request merged**